app/assets/javascripts/discourse/tests/acceptance/user-preferences-interface-test.js

```diff
@@ -310,5 +319,25 @@ acceptance(
         "resets dark scheme dropdown"
       );
     });
+
+    test("will not preview color scheme if it's not current user", async function (assert) {
+      let site = Site.current();
+
+      site.set("default_dark_color_scheme", { id: 1, name: "Dark" });
+      site.set("user_color_schemes", [
+        { id: 2, name: "Cool Breeze" },
+        { id: 3, name: "Dark Night", is_dark: true },
+      ]);
+
+      await visit("/u/charlie/preferences/interface");
+
+      await selectKit(".light-color-scheme .combobox").expand();
+      await selectKit(".light-color-scheme .combobox").selectRowByValue(3);
+
+      assert.notOk(
+        document.querySelector("link#cs-preview-light"),
```

Comment:
```suggestion
        exists("link#cs-preview-light"),
```

Reply:
I already tried it. The `exists` method is not working here. I think it's because we dynamically added it using the `document.body.appendChild` method.

Reply:
ahh icic. Do we have another test that is asserting that previewing a theme works? Otherwise, this test is going to be brittle by itself since the `id` attribute may change and this is simply asserting that a link with such id does not exist. The assertion will still pass even if the `id` attribute is wrong.

Reply:
Okay I added the test case for that case too.